Simplenote/PasskeyRegistrator.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import AuthenticationServices
+
+class PasskeyRegistrationControllerDelegate: NSObject, ASAuthorizationControllerDelegate {
+    var onCompletion: ((Result<PublicKeyCredentialRegistration, Error>) -> Void)?
+
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: any Error) {
+        onCompletion?(.failure(error))
+    }
+
+    public func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        guard let credential = authorization.credential as? PublicKeyCredentialRegistration else {
+            onCompletion?(.failure(PasskeyError.couldNotRequestRegistrationChallenge))
+            return
+        }
+
+        onCompletion?(.success(credential))
```

Comment:
There's a retain cycle between `PasskeyRegistrationControllerDelegate` and `PasskeyRegistrator`.

Two ways to break that:
- Assign `onCompletion = nil` in 3x spots, here
- OR clear `internalAuthControllerDelegate.onCompletion`